make_BAF_igv.py

```diff
@@ -4,57 +4,79 @@
             sys.exit("Single sample VCF support only. Input file {} is a multisample VCF".format(args.inputfile))
+
         sampleid = vcf_reader.samples[0]
-        output_file.write("#track type=igv name=BAF_{} color=0,100,0 altColor==0,100,0 graphType=points windowingFunction=none maxHeightPixels=100 viewLimits=0,100\n".format(sampleid))
-        output_file.write("chromosome\tstart\tend\tlocus\tbaf\n")
+
+        header = ("#track type=igv name=BAF_{} color=0,100,0 altColor==0,100,0 "\
+            "graphType=points windowingFunction=none maxHeightPixels=100 viewLimits=0,100"\
+            "\nchromosome\tstart\tend\tlocus\tbaf".format(sampleid)
+            )
+        if args.outputfile:
+            output_file.write("{}\n".format(header))
+        else:
+            print(header)
+        
         for record in vcf_reader:  # Parse all variant positions, and determine BAF if possible.
             chrom = record.CHROM
             end = record.POS
             start = end - 1
             locus = "{}:{}-{}".format(chrom, start, end)
-            try:
+
+            if 'DP' in record.genotype(sampleid).data._asdict():
                 dp = record.genotype(sampleid)['DP']
-            except:  # Skip ref/ref cals without AD field (this does exist)
-                print("Locus {} has no DP field, skipping variant position as BAF could not be determined".format(locus))
+            else: # Skip variant position as BAF could not be determined because DP is missing
                 continue
+
             variant_call = record.genotype(sampleid).is_variant  # Returns True if a variant-call, and thus not a reference-call
             if dp >= args.mindepth:
                 if variant_call == True: # Calculate ad and baf for variant-calls
                     ad = record.genotype(sampleid)['AD']
                     baf = (ad[1] / dp) * 100
                 elif variant_call == False: # Calculate ad and baf for reference-calls
-                    try:
+
+                    if 'AD' in record.genotype(sampleid).data._asdict():
                         ad = [record.genotype(sampleid)['AD']]
                         baf = ((dp - ad[0]) / (dp)) * 100
-                    except:  # Skip ref/ref cals without AD field (this does exist)
-                        print("Locus {} has no AD field, skipping variant position as BAF could not be determined".format(locus))
+                    else: # Skip variant position as BAF could not be determined because AP is missing
                         continue
+
                 else: # In case variant status is not known (no call or ./.)
                     continue
             else:  # If threshold if not met, skip position
                 continue
 
             if len(ad) <= 2:  # skip multiallelic sites
-                output_file.write("{chrom}\t{start}\t{end}\t{locus}\t{baf:.2f}\n".format(
+                variant_line = "{chrom}\t{start}\t{end}\t{locus}\t{baf:.2f}".format(
```

Comment:
did you forget the \n or is it not needed anymore?

Reply:
intentional:  print statement gives default \n, so not needed in string.

Reply:
Similar in 'header' string